ui/app/pages/send/send-content/send-amount-row/send-amount-row.component.js

```diff
@@ -26,12 +26,26 @@ export default class SendAmountRow extends Component {
     updateSendAmount: PropTypes.func,
     updateSendAmountError: PropTypes.func,
     updateGas: PropTypes.func,
+    maxModeOn: PropTypes.bool,
   }
 
   static contextTypes = {
     t: PropTypes.func,
   }
 
+  componentDidUpdate (prevProps) {
+    const { maxModeOn: prevMaxModeOn, gasTotal: prevGasTotal } = prevProps
+    const { maxModeOn, amount, gasTotal, selectedToken } = this.props
+
+    if (maxModeOn && selectedToken && !prevMaxModeOn) {
+      if (prevGasTotal !== gasTotal) {
```

Comment:
>Though...` validateAmount uses gasTotal, which hasn't changed at this point. It changes in a subsequent render, after the gas price is re-calculated.
Maybe there should be a separate block for handling validateAmount? e.g. if the gas total changes, call validateAmount again.

@Gudahtt Is this what you meant? I did it separately, but there was issues when coming back from the confirm tx screen via the `< Edit` and selecting a tx fee tab.

Reply:
I did mean separately - this would still only check for changes when `maxModeOn` changes, which I thought would be in a separate render :thinking: But clearly my suggestion didn't work...
I'll look into this a bit more. Maybe they are coming in on the same render after all.

Reply:
I tried what you described - coming back to the confirm screen via the `< Edit` - and the UI blew up for a different reason (fixed in #8182 )